pkg/cluster/cluster.go

```diff
@@ -269,7 +279,15 @@ func (c *Cluster) run(stopC <-chan struct{}) {
 					break
 				}
 				// TODO: we can't handle another upgrade while an upgrade is in progress
-				c.logger.Infof("spec update: from: %v to: %v", c.cluster.Spec, event.cluster.Spec)
+				oldSpec, err := json.MarshalIndent(c.cluster.Spec, "", "    ")
+				if err != nil {
+					c.logger.Warnf("failed to marshal cluster spec: %v", err)
```

Comment:
Yes, but it's not supposed to be fatal since this marshalling is only for logging purposes. So don't want to stop the operator via panic.